ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/JsonRpcHttpService.java

```diff
@@ -148,6 +156,29 @@ private void validateConfig(final JsonRpcConfiguration config) {
     return resultFuture;
   }
 
+  private Handler<RoutingContext> checkWhitelistHostHeader() {
+    return event -> {
+      Optional<String> hostHeader =
+          Optional.ofNullable(Iterables.get(Splitter.on(':').split(event.request().host()), 0));
+      if ((config.hostWhitelist().size() == 1 && config.hostWhitelist().contains("*"))
```

Comment:
Is this a validation that when `*` is used that there is also only one entry in the whitelist?

Reply:
Yea, figured it'd be poor practice to define a wildcard _and_ explicit hostnames.  The documentation would specify that `*` can only be used by itself, to ensure people understand what they're doing.

Thoughts on my reasoning?

Reply:
Apparently Jake is the person to raise this with. 👍

Reply:
My point was going to be a user experience perspective one. It'd be smoother to perform the validation at the earliest point and fail with an appropriate error (i.e. using * with another input), rather than at this point in the code ...as it might not be clear to the user what they have done to cause the failure.

Reply:
I see, I'll see if we can move this logic back to the CLI parameter validation.